src/tests/runner/test_reporter.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+* (C) 2022 Jack Lloyd
+* (C) 2022 René Meusel, Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include "test_reporter.h"
+
+#include <numeric>
+
+namespace Botan_Tests {
+
+Reporter::Reporter(const Test_Options& opts)
+   : m_total_test_runs(opts.test_runs())
+   , m_current_test_run(0)
+   {}
+
+void Reporter::set_property(const std::string& name, const std::string& value)
+   {
+   m_properties.insert_or_assign(name, value);
+   }
+
+void Reporter::next_test_run()
+   {
+   m_start_time = std::chrono::high_resolution_clock::now();
+   ++m_current_test_run;
+
+   next_run();
+   }
+
+void Reporter::record(const std::string& testsuite_name,
+                      const std::vector<Botan_Tests::Test::Result>& results)
+   {
+   // TODO: Is that still required or could it be modernized?
+   std::map<std::string, Botan_Tests::Test::Result> combined;
+   for(auto const& result : results)
+      {
+      const std::string who = result.who();
+      auto i = combined.find(who);
+      if(i == combined.end())
+         {
+         combined.insert(std::make_pair(who, Botan_Tests::Test::Result(who)));
+         i = combined.find(who);
+         }
+
+      i->second.merge(result);
+      }
```

Comment:
I copied that verbatim from the old `Test_Result_State::record()` method without fully grasping what it's supposed to achieve. @randombit any idea?

Reply:
IIRC this is to combine multiple tests that have the same summary within a single test. Eg if two `Test::Result("foo")` are created this has the effect of combining them into a single report in the final output.